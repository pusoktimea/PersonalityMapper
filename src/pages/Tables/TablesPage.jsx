import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import cx from 'classnames';

import Loader from 'components/Loader';
import Panel from 'components/Panel';
import Pagination from 'components/Pagination';
import Button from 'components/Button';
import Table from 'components/Table';
import TableColumn from 'components/Table/TableColumn';

import './tables-page.scss';

class TablesPage extends PureComponent {
  static propTypes = {
    isTablesLoaded: PropTypes.bool.isRequired,
    onGetTables: PropTypes.func.isRequired,
    tables: PropTypes.object.isRequired,
    isSideBarMinimised: PropTypes.bool
  }

  componentWillMount() {
    if (!this.props.isTablesLoaded) {
      this.props.onGetTables();
    }
  }

  render() {
    const {
      tables: {
        list,
        pagination_info
      },
      isTablesLoaded,
      isSideBarMinimised
    } = this.props;
    const baseClass = 'main-content';

    return (
      <div className={cx('tables-page', baseClass, isSideBarMinimised && `${baseClass}--stretched`)}>
        <h2 className="title">Tables</h2>
        {
          !isTablesLoaded ?
            <Loader /> :
            <Panel title="App Tables">
              <Table
                bordered={false}
                hovered={false}
                striped={false}
                selectable={false}
                items={list}
              >
                <TableColumn
                  header="Table Name"
                  cellClassName="table-name-cell"
                  contentGetter={(item) => (
                    <Button
                      theme="default"
                      href={`/tables/${item.name}`}
                    >
                      {item.name.replace(/_/g, ' ')}
                    </Button>
                  )}
                />
                <TableColumn
                  header="Created"
                  cellClassName="creation-date-cell"
                  contentGetter={(item) => (
                    moment(item.create_adt * 1000).format('MMM DD, YYYY')
                  )}
                />
                <TableColumn
                  header="Records"
                  cellClassName="row-count-cell"
                  contentGetter="rows"
                />
              </Table>
              {pagination_info.total_pages > 1 && (
                <Pagination totalPages={pagination_info.total_pages} />
              )}
            </Panel>
        }
      </div>
    );
  }
}

export default TablesPage;
