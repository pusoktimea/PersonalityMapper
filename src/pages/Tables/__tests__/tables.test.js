import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';

import TablesPage from '../TablesPage';

const pageProps = {
  isTablesLoaded: true,
  onGetTables: jest.fn(),
  tables: {
    list: [{
      name: 'table 1',
      id: 'table_1',
      rows: 100,
      create_adt: 1503818195
    }, {
      name: 'table 2',
      id: 'table_2',
      rows: 200,
      create_adt: 1503818195
    }],
    pagination_info: {
      total_pages: 1
    }
  },
  isSideBarMinimised: false
};

describe('Tables Page test', () => {
  it('renders and matches snapshot', () => {
    expect(
      renderer.create(
        <StaticRouter context={{}}>
          <TablesPage {...pageProps} />
        </StaticRouter>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
