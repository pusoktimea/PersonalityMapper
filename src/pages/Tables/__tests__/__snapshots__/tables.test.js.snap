// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tables Page test renders and matches snapshot 1`] = `
<div
  className="tables-page main-content"
>
  <h2
    className="title"
  >
    Tables
  </h2>
  <div
    className="activate-panel"
  >
    <div
      className="activate-panel_header"
    >
      App Tables
    </div>
    <div
      className="activate-panel_body"
    >
      <table
        className="activate-table activate-table--condensed"
      >
        <thead
          className="activate-table_header"
        >
          <tr>
            <th
              className="activate-table_header_item"
            >
              Table Name
            </th>
            <th
              className="activate-table_header_item"
            >
              Created
            </th>
            <th
              className="activate-table_header_item"
            >
              Records
            </th>
          </tr>
        </thead>
        <tbody
          className="activate-table_body"
        >
          <tr
            className="activate-table_row activate-table_row--odd"
            onClick={[Function]}
          >
            <td
              className="activate-table_row_column table-name-cell"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default"
                disabled={false}
                href="/tables/table 1"
                onClick={[Function]}
              >
                table 1
              </a>
            </td>
            <td
              className="activate-table_row_column creation-date-cell"
            >
              Aug 27, 2017
            </td>
            <td
              className="activate-table_row_column row-count-cell"
            >
              100
            </td>
          </tr>
          <tr
            className="activate-table_row"
            onClick={[Function]}
          >
            <td
              className="activate-table_row_column table-name-cell"
            >
              <a
                className="activate-button activate-button--default activate-button--size-default"
                disabled={false}
                href="/tables/table 2"
                onClick={[Function]}
              >
                table 2
              </a>
            </td>
            <td
              className="activate-table_row_column creation-date-cell"
            >
              Aug 27, 2017
            </td>
            <td
              className="activate-table_row_column row-count-cell"
            >
              200
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
